lib/DependencyScan/ModuleDependencyCacheSerialization.cpp

```diff
@@ -802,52 +802,55 @@ unsigned Serializer::getArray(ModuleDependencyID moduleID,
-    // Add the module's name
-    addIdentifier(moduleID.first);
-    // Add the module's dependencies
-    addArray(moduleID, ModuleIdentifierArrayKind::DirectDependencies,
-             dependencyInfo->getModuleDependencies());
-
-    // Add the dependency-kind-specific data
-    if (auto swiftTextDeps = dependencyInfo->getAsSwiftTextualModule()) {
-      if (swiftTextDeps->swiftInterfaceFile)
-        addIdentifier(swiftTextDeps->swiftInterfaceFile.getValue());
-      addArray(moduleID, ModuleIdentifierArrayKind::CompiledModuleCandidates,
-               swiftTextDeps->compiledModuleCandidates);
-      addArray(moduleID, ModuleIdentifierArrayKind::BuildCommandLine,
-               swiftTextDeps->buildCommandLine);
-      addArray(moduleID, ModuleIdentifierArrayKind::ExtraPCMArgs,
-               swiftTextDeps->extraPCMArgs);
-      addIdentifier(swiftTextDeps->contextHash);
-      if (swiftTextDeps->bridgingHeaderFile)
-        addIdentifier(swiftTextDeps->bridgingHeaderFile.getValue());
-      addArray(moduleID, ModuleIdentifierArrayKind::SourceFiles,
-               swiftTextDeps->sourceFiles);
-      addArray(moduleID, ModuleIdentifierArrayKind::BridgingSourceFiles,
-               swiftTextDeps->bridgingSourceFiles);
-      addArray(moduleID, ModuleIdentifierArrayKind::BridgingModuleDependencies,
-               swiftTextDeps->bridgingModuleDependencies);
-    } else if (auto swiftBinDeps = dependencyInfo->getAsSwiftBinaryModule()) {
-      addIdentifier(swiftBinDeps->compiledModulePath);
-      addIdentifier(swiftBinDeps->moduleDocPath);
-      addIdentifier(swiftBinDeps->sourceInfoPath);
-    } else if (auto swiftPHDeps =
-                   dependencyInfo->getAsPlaceholderDependencyModule()) {
-      addIdentifier(swiftPHDeps->compiledModulePath);
-      addIdentifier(swiftPHDeps->moduleDocPath);
-      addIdentifier(swiftPHDeps->sourceInfoPath);
-    } else if (auto clangDeps = dependencyInfo->getAsClangModule()) {
-      addIdentifier(clangDeps->moduleMapFile);
-      addIdentifier(clangDeps->contextHash);
-      addArray(moduleID, ModuleIdentifierArrayKind::NonPathCommandLine,
-               clangDeps->nonPathCommandLine);
-      addArray(moduleID, ModuleIdentifierArrayKind::FileDependencies,
-               clangDeps->fileDependencies);
-    } else {
-      llvm_unreachable("Unexpected module dependency kind.");
+    auto dependencyInfos = cache.findAllDependenciesIrrespectiveOfSearchPaths(
+        moduleID.first, moduleID.second);
+    assert(dependencyInfos.hasValue() && "Expected dependency info.");
+    for (auto &dependencyInfo : *dependencyInfos) {
+      // Add the module's name
+      addIdentifier(moduleID.first);
+      // Add the module's dependencies
+      addArray(moduleID, ModuleIdentifierArrayKind::DirectDependencies,
+               dependencyInfo.getModuleDependencies());
+
+      // Add the dependency-kind-specific data
+      if (auto swiftTextDeps = dependencyInfo.getAsSwiftTextualModule()) {
```

Comment:
Instead of using chaining `if-else` here, can we refactor it to a switch statement?

Reply:
Done, thanks for pointing it out.